### ISSUE TITLE
Python: Provide Stream.punctor()

### ISSUE THREAD

**Issue by markheger** [IBMStreams/streamsx.topology#2517]
Support to emit Window punctuation like the SPL Punctor, but Python func is called to determine the condition when punct needs to be sent.

Stream.print() has a new parameter to print the punctuations to stdout.

```
topo = Topology()
s = topo.source([1,2,3,4])
s = s.punctor(lambda t : 2 < t)
s.print(write_punctuations=True)
```

**Comment by markheger** [comment]
merged to IBMStreams:feature/punct

**Comment by markheger** [comment]
Add an option that Punctor replaces a tuple by a window punctuation.

**Comment by markheger** [comment]
```
        topo = Topology()
        s = topo.source(generate_numbers_for_named_tuple_schema)
        s = s.punctor(func=(lambda t : True == t.punct_flag), replace=True)
        s = s.map(lambda x : (x.value,), schema='tuple<int32 z>')
        s.print(write_punctuations=True)
```

**Comment by markheger** [comment]
merged to develop branch for 1.16.0a
